### ISSUE TITLE
LastfmError[500]: 500 Internal Server Error (lastfm.User.GetRecentTracks)

### ISSUE THREAD

**Issue by skupjoe** [dikey0ficial/dlfm#2]
```
$ ./DiscordLastfmScrobbler -C ./config.ini -d
2021/12/10 00:27:54 Settings loaded: config.ini
2021/12/10 00:27:54 Endless mode! Ctrl+C to exit
2021/12/10 00:27:54 Authorized to Discord
2021/12/10 00:27:55 Connected to Discord
2021/12/10 02:28:08 LastFM error:  LastfmError[500]: 500 Internal Server Error (lastfm.User.GetRecentTracks)
2021/12/10 06:56:49 LastFM error:  LastfmError[500]: 500 Internal Server Error (lastfm.User.GetRecentTracks)
2021/12/10 08:22:06 LastFM error:  LastfmError[500]: 500 Internal Server Error (lastfm.User.GetRecentTracks)
2021/12/10 10:27:35 Discord error:  no websocket connection exists
2021/12/10 11:17:13 LastFM error:  LastfmError[500]: 500 Internal Server Error (lastfm.User.GetRecentTracks)
2021/12/10 11:34:40 LastFM error:  LastfmError[500]: 500 Internal Server Error (lastfm.User.GetRecentTracks)
```

I recall I was seeing this 500 error with [DiscordLastfmScrobbler](https://github.com/edvardpotter/DiscordLastfmScrobbler) version as well. So, I think this has been a longstanding issue.

My config has `check_interval = 7` and `endless_mode = true`.

You can see that I also get `Discord error:  no websocket connection exists` every so often, as well.

**Comment by skupjoe** [comment]
> But I have one question. What are `-C ./config.ini` and `-d` arguments in `./DiscordLastfmScrobbler -C ./config.ini -d`

Old copy-paste, sorry. I am not using these flags anymore and still seeing the issue.

**Comment by dikey0ficial** [comment]
ok... Can you give me information about your system? maybe this will help

**Comment by skupjoe** [comment]
I experienced this both on Windows and Linux (CentOS 7), so it is not machine-specific, AFAICT.

**Comment by dikey0ficial** [comment]
Ok. If you have other computer, can you try it on them and write result?
Also, can you connect to last.fm from browser? So you use proxy or VPN? Do you have static IP or dynamic?
Also, try to generate new token, maybe it will help
I'm sorry that I spend your time with questions, but it really hard to understand where is bug :)

**Comment by skupjoe** [comment]
> Ok. If you have other computer, can you try it on them and write result? Also, can you connect to last.fm from browser? So you use proxy or VPN? Do you have static IP or dynamic? Also, try to generate new last.fm API token, maybe it will help I'm sorry that I spend your time with questions, but it really hard to understand where is bug :)

Indeed, it is a strange/intermittent issue. For the last 12 hours, I have not received any error. But earlier this morning (while I was sleeping) I see that it was observed multiple times back-to-back.

I've reproduced this from a Windows computer and a Linux VM. Both using same IP/network. I am not passing the traffic through a proxy or IP. I can access lastfm.com from a browser fine (right now), but I am not sure if I can access during these periods- it is hard to catch in-time :)

**Comment by skupjoe** [comment]
I've generated a new API token and I am running dlfm on an ec2 instance now as another test. I will let you know if I experience the same.

Another thought- maybe 7 seconds interval is too fast and lastfm is rate-limiting me? You've never experienced this error?

**Comment by dikey0ficial** [comment]
> Another thought- maybe 7 seconds interval is too fast and lastfm is rate-limiting me? You've never experienced this error?

Maybe yes, you should try bigger interval. I've never seen this error and I haven't reproduced it.

Also, if you have problems with a new token too, try to register new account and generate token from it, but don't change nickname in config, this also can help.

**Comment by dikey0ficial** [comment]
@skupjoe, have error reproduced again since last comment?

**Comment by skupjoe** [comment]
@dikey0ficial Looks like I am still seeing the same error on ec2 instance using your latest source:

![image](https://user-images.githubusercontent.com/5000412/148028528-8a2e02b1-f5ec-465d-a927-32911d07a3a7.png)

I will try increasing the polling interval to 20 seconds.

**Comment by skupjoe** [comment]
Still an issue with `check_interval` set to 20 seconds:

![image](https://user-images.githubusercontent.com/5000412/148102954-ef3f7602-58b4-4b48-b6d5-1a5b77dbd4f7.png)

I set this config and went to bed. Woke up with these logs above.

**Comment by dikey0ficial** [comment]
> Still an issue with `check_interval` set to 20 seconds:
> 
> ![image](https://user-images.githubusercontent.com/5000412/148102954-ef3f7602-58b4-4b48-b6d5-1a5b77dbd4f7.png)
> 
> I set this config and went to bed. Woke up with these logs above.

Yesterday I had the same error, but after a minute this stopped. I don't know why do we have this error, maybe it's error on last.fm side, but I'll check all API calls

**Comment by dikey0ficial** [comment]
@skupjoe i've just updated. may be this build will work?

P.S. In your screenshots from previous comments there is your last,fm token. I recomend you to delete this one or edit images and update them :)

**Comment by skupjoe** [comment]
🤦‍♂️ Thanks I've disabled that API key and made a new one.

I'll try out your new build and lyk- thanks.

**Comment by skupjoe** [comment]
Still exhibited on 4.0.0 but I'm now running on 4.2.2.2 and increased my update interval to 20 seconds. wlyk
